File: FLUTTER/igest_app/lib/models/principal/empresa_sistema.dart
class EmpresaSistema {
  int? idEmpSis;
  int? idEmpresa;
  String? idSistema;
  bool? activo;
  bool? soloLectura;
  int? idTipoEdition;

  EmpresaSistema({
    this.idEmpSis,
    this.idEmpresa,
    this.idSistema,
    this.activo,
    this.soloLectura,
    this.idTipoEdition,
  });

  EmpresaSistema.fromJson(Map<String, dynamic> json)
      : idEmpSis = json['idEmpSis'],
        idEmpresa = json['idEmpresa'],
        idSistema = json['idSistema'],
        activo = json['activo'],
        soloLectura = json['soloLectura'],
        idTipoEdition = json['idTipoEdition'];

  Map<String, dynamic> toJson() => {
        'idEmpSis': idEmpSis,
        'idEmpresa': idEmpresa,
        'idSistema': idSistema,
        'activo': activo,
        'soloLectura': soloLectura,
        'idTipoEdition': idTipoEdition,
      };

  EmpresaSistema copyWith({
    int? idEmpSis,
    int? idEmpresa,
    String? idSistema,
    bool? activo,
    bool? soloLectura,
    int? idTipoEdition,
  }) {
    return EmpresaSistema(
      idEmpSis: idEmpSis ?? this.idEmpSis,
      idEmpresa: idEmpresa ?? this.idEmpresa,
      idSistema: idSistema ?? this.idSistema,
      activo: activo ?? this.activo,
      soloLectura: soloLectura ?? this.soloLectura,
      idTipoEdition: idTipoEdition ?? this.idTipoEdition,
    );
  }
}

File: SPRING/igest.server/src/main/java/com/yupanastudio/igest/server/principal/model/EmpresaSistema.java
/*
 * Document   : EmpresaSistema
 * Author     : Franklin Barzola Olivares <frankoantonio@gmail.com>
 */
package com.yupanastudio.igest.server.principal.model;

import org.springframework.data.annotation.Id;

public class EmpresaSistema {

    @Id
    private Integer idEmpSis;
    private int idEmpresa;
    private String idSistema;
    private Boolean activo;
    private Boolean soloLectura;
    private int idTipoEdition;

    public Integer getIdEmpSis() {
        return idEmpSis;
    }

    public void setIdEmpSis(Integer idEmpSis) {
        this.idEmpSis = idEmpSis;
    }

    public int getIdEmpresa() {
        return idEmpresa;
    }

    public void setIdEmpresa(int idEmpresa) {
        this.idEmpresa = idEmpresa;
    }

    public String getIdSistema() {
        return idSistema;
    }

    public void setIdSistema(String idSistema) {
        this.idSistema = idSistema;
    }

    public Boolean getActivo() {
        return activo;
    }

    public void setActivo(Boolean activo) {
        this.activo = activo;
    }

    public Boolean getSoloLectura() {
        return soloLectura;
    }

    public void setSoloLectura(Boolean soloLectura) {
        this.soloLectura = soloLectura;
    }

    public int getIdTipoEdition() {
        return idTipoEdition;
    }

    public void setIdTipoEdition(int idTipoEdition) {
        this.idTipoEdition = idTipoEdition;
    }

}
